src/iuse.cpp

```diff
@@ -6459,20 +6559,800 @@ struct npc_photo_def : public JsonDeserializer, public JsonSerializer {
+        return name;
+    }
+    for( const std::string &ter_good : ter_whitelist ) {
+        if( ter->id.c_str() == ter_good ) {
+            return name;
+        }
+    }
+    for( const std::string &flag : flags ) {
+        if( ter->has_flag( flag ) ) {
+            return name;
+        }
+    }
+
+    return std::string();
+}
+
+static std::string colorized_feature_description_at( const tripoint center_point, bool &item_found,
+        const units::volume &min_visible_volume )
+{
+    item_found = false;
+    const furn_id furn = g->m.furn( center_point );
+    if( furn != f_null && furn.is_valid() ) {
+        std::string furn_str = "<color_yellow>" + furn->name() + "</color>";
+        std::string sign_message = g->m.get_signage( center_point );
+        if( !sign_message.empty() ) {
+            furn_str += _( " with message \"" ) + sign_message + "\"";
+        }
+        if( !furn->has_flag( "CONTAINER" ) && !furn->has_flag( "SEALED" ) ) {
+            const item item = get_top_item_at_point( center_point, min_visible_volume );
+            if( !item.is_null() ) {
+                furn_str += _( " with " ) + colorized_item_name( item );
+                item_found = true;
+            }
+        }
+        return furn_str;
+    }
+    return std::string();
+}
+
+static std::string get_vehicle_hash( const vehicle &veh )
+{
+    const tripoint veh_coord = veh.global_pos3();
+    return string_format( "%s %i %i %i", veh.disp_name(), veh_coord.x, veh_coord.y, veh_coord.z );
+}
+
+static const auto format_object_pair = []( const std::pair<std::string, int> &pair,
+                                       const std::string &article = "" )
+{
+    if( pair.second == 1 ) {
+        return string_format( "%s%s", article, pair.first );
+    } else if( pair.second > 1 ) {
+        return string_format( "%s%i %s", article, pair.second, pair.first );
+    }
+    return std::string();
+};
+static const auto format_object_pair_article = []( const std::pair<std::string, int> &pair )
+{
+    return format_object_pair( pair, "a " );
+};
+
+static std::string effects_description_for_creature( Creature *const creature, std::string &pose,
+        const std::string &pronoun_sex )
+{
+    struct ef_con { // effect constraint
+        std::string status;
+        std::string pose;
+        int intensity_lower_limit;
+        ef_con( std::string status, std::string pose, int intensity_lower_limit ) :
+            status( status ), pose( pose ), intensity_lower_limit( intensity_lower_limit ) {};
+        ef_con( std::string status, std::string pose ) :
+            status( status ), pose( pose ), intensity_lower_limit( 0 ) {};
+        ef_con( std::string status, int intensity_lower_limit ) :
+            status( status ), pose( "" ), intensity_lower_limit( intensity_lower_limit ) {};
+        ef_con( std::string status ) :
+            status( status ), pose( "" ), intensity_lower_limit( 0 ) {};
```

Comment:
And the semicolon at the end of a function definition is not needed.
```suggestion
            status( status ), pose( "" ), intensity_lower_limit( 0 ) {}
```